### PR TITLE
Add Dict.deleteKey, Dict.deleteKeys and Dict.selectKeys

### PR DESCRIPTION
**Pull request by anthony-khong**
Hi @mobily, please let me know if you'd like me to change anything at all here!

### CODE REVIEW

File: __tests__/Dict/selectKeys.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  it('*', () => {
+    expect(
+      pipe(
+        { name: 'Joe', location: 'Warsaw', age: 42 },
+        D.selectKeys(['name', 'age']),
+      ),
+    ).toEqual({ name: 'Joe', age: 42 })
+    expect(
+      pipe(
+        { name: 'Joe', location: 'Warsaw' } as Record<string, string>,
```
Comment (mobily) [inline __tests__/Dict/selectKeys.test.ts:60]: is `Record<string, string>` necessary here? if so, I feel we need to work on types and improve them somehow, I will look into it :)

Comment (anthony-khong) [reply __tests__/Dict/selectKeys.test.ts:60]: Hi @mobily! What I'm trying to get at here is that if the type of `{ name: 'Joe', location: 'Warsaw'}` is `{ name: string, location: string}`, selecting `age` would not type check. This situation can arise if you are dealing with records - if the key is not present, you should not get anything selected. Perhaps the intent of the test is not super clear! Please let me know how you'd like to change it!

Comment (mobily) [reply __tests__/Dict/selectKeys.test.ts:60]: ahh, now I got it, thanks for the explanation! ❤️ you may want to check out how I handle this case:
https://github.com/mobily/ts-belt/blob/master/__tests__/Dict/update.test.ts#L115

**Comment by mobily** [comment]
@anthony-khong thanks again for your contribution! could you resolve conflicts, please? in general, it looks really good, and I believe we can merge it once it's ready for merging 🚀

**Comment by anthony-khong** [comment]
I've resolved the conflicts, changed the examples and verified that the tests are still good. Thank you for reviewing the PR, please let me know if you'd like me to change anything else!

**Review by mobily — APPROVED** [review]

**Comment by mobily** [comment]
thank you @anthony-khong, this looks awesome! ❤️

**Pull request merged**